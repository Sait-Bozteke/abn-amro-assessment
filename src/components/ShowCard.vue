<script lang="ts">
import { defineComponent, PropType, computed } from 'vue';
import { Show } from '../types/Show';

export default defineComponent({
  props: {
    show: {
      type: Object as PropType<Show>,
      required: true,
    },
  },

  setup(props) {
    const ratingClass = computed(() => {
      const rating = props.show.rating.average;
      if (rating === null || rating === undefined) return 'bg-secondary';
      if (rating < 6) return 'bg-danger';
      if (rating >= 6 && rating <= 8) return 'bg-warning text-dark';
      if (rating > 8) return 'bg-success';
      return 'bg-secondary';
    });

    return {
      ratingClass,
    };
  },
});
</script>
