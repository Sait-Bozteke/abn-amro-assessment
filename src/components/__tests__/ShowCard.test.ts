import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import ShowCard from '@/components/ShowCard.vue';
import { Show } from '../../types/Show';

const show: Show = {
  id: 1,
  name: 'Test Show',
  image: {
    original: 'http://example.com/image.jpg',
    medium: '',
  },
  genres: ['Drama', 'Action'],
  rating: { average: 7.5 },
  summary: 'This is a test summary.',
};

describe('ShowCard.vue', () => {
  it('renders correctly and matches the snapshot', () => {
    const wrapper = shallowMount(ShowCard, {
      props: {
        show,
      },
      global: {
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders props.show correctly', () => {
    const wrapper = shallowMount(ShowCard, {
      props: {
        show,
      },
      global: {
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
    });

    const img = wrapper.find('img');
    expect(img.exists()).toBe(true);
    expect(img.attributes('src')).toBe(show.image.original);

    const title = wrapper.find('.card-title');
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe(show.name);

    const genres = wrapper.find('.card-text');
    expect(genres.exists()).toBe(true);
    expect(genres.text()).toContain('Genres: Drama, Action');

    const rating = wrapper.find('.badge');
    expect(rating.exists()).toBe(true);
    expect(rating.text()).toBe(show.rating.average.toString());

    const detailsLink = wrapper.findComponent(RouterLinkStub);
    expect(detailsLink.exists()).toBe(true);
    expect(detailsLink.props('to')).toEqual({
      name: 'ShowDetails',
      params: { id: show.id },
    });
  });
});
