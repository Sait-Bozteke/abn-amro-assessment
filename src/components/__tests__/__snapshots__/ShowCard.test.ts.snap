// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCard.vue renders correctly and matches the snapshot 1`] = `
"<div class="card"><img src="http://example.com/image.jpg" class="card-img-top" alt="Show Image">
  <div class="card-body">
    <h5 class="card-title">Test Show</h5>
    <p class="card-text">Genres: Drama, Action</p>
    <p class="card-text"><strong>Rating:</strong><span class="bg-warning text-dark badge rounded-pill">7.5</span></p><a class="btn btn-primary"> Details </a>
  </div>
</div>"
`;
